File: 74-search-a-2d-matrix/search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        rows = len(matrix)
        cols = len(matrix[0])
        l=0
        r=(rows*cols)-1
        while l <= r:
            mid = l + (r-l) // 2
            rx = mid//cols
            cx = mid%cols
            if matrix[rx][cx] == target:
                return True
            elif matrix[rx][cx] > target:
                r = mid - 1
            else:
                l = mid+1

        return False

File: 74-search-a-2d-matrix/search-a-2d-matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int m = matrix.size();
        int n = matrix[0].size();
        int total = m*n;
        int l =0, r = total-1;
        while(l<=r){
            int mid = l + ((r-l)/2);
            int rx=mid/n, cx=mid%n;
            if (matrix[rx][cx]==target) return true;
            else if (matrix[rx][cx]>target) r=mid-1;
            else l=mid+1;
        }
        return false;
    }
};